pkg/operator/certrotation/signer_test.go

```diff
@@ -1,24 +1,216 @@
 package certrotation
 
 import (
+	"bytes"
 	"context"
+	"reflect"
 	"strings"
+	"sync"
 	"testing"
 	"time"
 
 	"github.com/davecgh/go-spew/spew"
 
 	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	kubefake "k8s.io/client-go/kubernetes/fake"
 	corev1listers "k8s.io/client-go/listers/core/v1"
 	clienttesting "k8s.io/client-go/testing"
 	"k8s.io/client-go/tools/cache"
 
 	"github.com/openshift/api/annotations"
+	"github.com/openshift/library-go/pkg/crypto"
 	"github.com/openshift/library-go/pkg/operator/events"
 )
 
+// loadbalancer-serving-signer secret in openshift-kube-apiserver-operator namespace
+// is reconciled by multiple controllers at the same time without any coordination.
+//
+// the following unit test demonstrates that the secret crypto material
+// can be regenerated, which has serious consequences for the platform
+// as it can break external clients and the cluster itself.
+//
+// controllers reconciling the secret:
```

Comment:
it looks like we have `2` separate processes and `4` different controllers competing over the same resource.